pkg/net/libp2p/key_test.go

```diff
@@ -23,25 +25,25 @@ func TestOperatorPrivateKeyToNetworkKeyPair(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if !reflect.DeepEqual(networkPublicKey.Curve.Params(), DefaultCurve.Params()) {
-		t.Errorf("network public key uses the wrong curve")
-	}
+	btcecPublicKey := (*btcec.PublicKey)(networkPublicKey)
```

Comment:
Worth dropping a comment explaining that the `*libp2pcrypto.Secp256k1PublicKey` does not expose an easy way to get X and Y coords so we need to convert the key to the underlying `*btcec.PublicKey` in order to get what we need.

Reply:
Done in 58c5230ded309eb47e97dd595427798f6ac0d71d.